cub/cub/block/block_merge_sort.cuh

```diff
@@ -94,8 +94,13 @@ _CCCL_DEVICE _CCCL_FORCEINLINE void SerialMerge(
   const int keys1_end = keys1_beg + keys1_count;
   const int keys2_end = keys2_beg + keys2_count;
 
-  KeyT key1 = keys_shared[keys1_beg];
-  KeyT key2 = keys_shared[keys2_beg];
+  // Either range may be empty. If it is, load a valid key from the other range
+  // to avoid out-of-bounds access:
+  const int key1_initial_load_offset = (keys1_count != 0) ? keys1_beg : keys2_beg;
+  const int key2_initial_load_offset = (keys2_count != 0) ? keys2_beg : keys1_beg;
```

Comment:
If either range is empty, should we explicitly guard against *both* being empty or is that handled elsewhere?

Reply:
Higher level logic ensures that at least one range will be valid.